enforcer_cached.go

```diff
@@ -93,3 +93,36 @@ func (e *CachedEnforcer) setCachedResult(key string, res bool) {
 func (e *CachedEnforcer) InvalidateCache() {
 	e.m = make(map[string]bool)
 }
+
+/* Clear cache after update Policy*/
+// AddPolicy adds an authorization rule to the current policy.
+// If the rule already exists, the function returns false and the rule will not be added.
+// Otherwise the function returns true by adding the new rule.
+func (e *CachedEnforcer) AddPolicy(params ...interface{}) (bool, error) {
+	ok, err := e.AddNamedPolicy("p", params...)
+	e.InvalidateCache()
```

Comment:
also please check `if ok`, avoid unecessary clear is important

Reply:
sure